src/awesome_map/AwesomeMap.js

```diff
@@ -148,6 +148,18 @@ define(function(require) {
         this.onScaleChanged = Observable.newObservable();
 
         /**
+         * Observable for subscribing to pending scale changes.
+         *
+         * @method AwesomeMap#onScaleWillChange
+         * @param {Function} callback
+         *        Invoked with (sender, {
+         *            currentScale: {number}
+         *            nextScale: {number}
+         *        })
+         */
+        this.onScaleWillChange = Observable.newObservable();
```

Comment:
Interceptors rely on being able to do this during `onTransformStarted`, so yeah, I think it's fine.